ChangeLog

```diff
@@ -1,3 +1,167 @@
+	  (CFE-2084)
+	- Improve support for Alpine Linux
+	- Set the exit value when running cf-key.
+	  When running cf-key to generate new keys, set the exit value of the
+	  program to be 0 on success and 1 on failure. This makes it easier to
+	  catch errors during setup of a new machine.
+	  Change the default behavior of the program to not write anything to stdout,
+	  opting to use the Log() function which can write to stdout and will also
+	  allow output to be sent to syslog.
+	  Add a --inform option to set the global log level to LOG_LEVEL_INFO.
+	  Change the permissions of the randseed file to 600 and catch the exception
+	  if the chmod call fails.
+	- Properly reverse-resolve DNS names longer than 63 chars (ENT-3379)
+	- Properly redirect init script to systemd on debian systems (ENT-3326)
+
+	Bug fixes:
+	- Disallow modifications of variables from a remote bundle (CFE-1915)
+	- Speedup evalution by not copying variables table when expanding a promise
+	  (CFE-2524)
+	- Resolve subkey conflicts when converting to JSON
+	  Whenever there is a conflict of array variable definitions prefer
+	  the container subkeys over simple values when converting to JSON
+	  (CFE-2536)
+	- Do not ignore meta promises in server bundles (CFE-2066)
+	- Add a debug log for computed class in splayclass
+	- Don't error when calling isexecutable on broken link (CFE-741)
+	- Fix segfault when no show-evaluated-vars/classes is specified
+	- Fix memory leak in cf-execd, triggered when sending email failed (CFE-2712)
+	- fix IPv6 parsing to be un-reversed (CFE-2580)
+	- Fix bug preventing permission changes on Unix sockets (CFE-1782)
+	- fix storage mount promise when existing mountpoint has a similar path
+	  (CFE-1960)
+	- Fix segfault when cf-promises -p is called against a file with syntax
+	  errors (CFE-2696)
+	- Fix rare cf-execd hang (CFE-2719)
+	- Fix mergedata segfault when called on a non-container (CFE-2704)
+	- Do not segfault if policy_server.dat only contains whitespaces and/or line breaks
+	- Fix segfault on JSON policy files with no bundles and bodies (CFE-2754)
+
+
+3.11.0:
+	New Features:
+	- allow function calls in promiser using universal "with" attribute.
+	  (CFE-1092)
+	- add example of with attribute (CFE-1092)
+	- Detect Amazon Linux and set "AmazonLinux" hard class and
+	  sys.flavour variable.
+	- New sysctlvalue() and data_sysctlvalues() functions from /proc/sys
+	  (CFE-2513)
+	- readdata() also auto-detects .yml files as YAML
+	- Added support for ENV and CSV file parsing (CFE-1881)
+	- Added vars and classes for CoreOS (ENT-3043)
+	- cf-agent: implement --show-evaluated-vars and --show-evaluated-classes
+	- Support for custom ports and host names as policy hub (CFE-953)
+	- cf-promises: allows --show-vars and --show-classes to take an optional filter
+	- Added a new tool: cf-net. cf-net is a CLI for the CFEngine
+	  network protocol, useful for debugging, testing etc.
```

Comment:
I would use this description of cf-net in the New Features section above about cf-net.

Reply:
hm, but did cf-net appeared first in 3.11 or 3.12? According to 3.11 changelog, it was already there. Then want is it doing in 3.12 changelog? Or do we want to remove it from 3.11 changelog, in order to "promote" by putting to the top of changelog file?

Reply:
In your current changelog it appears under "New Features" in 3.12 and regular changelog of 3.11. I though this was to highlight it in 3.12, I just disliked the description there. I suggest using this description in both places.

Reply:
ok!